src/main/java/com/kodilla/ecommercee/controller/UserController.java

```diff
@@ -1,44 +1,38 @@
 package com.kodilla.ecommercee.controller;
 
+import com.kodilla.ecommercee.controller.exceptions.UserNotFoundException;
+import com.kodilla.ecommercee.domain.users.GuavaCacheService;
 import com.kodilla.ecommercee.domain.users.UserDto;
+import com.kodilla.ecommercee.mapper.UserMapper;
+import com.kodilla.ecommercee.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Random;
+import java.util.concurrent.ExecutionException;
 
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 
 @RestController
 @RequestMapping("/v1/user")
 public class UserController {
-    private List<UserDto> users = new ArrayList<>();
+
+    @Autowired
+    private UserService userService;
+    @Autowired
+    private UserMapper userMapper;
 
     @PostMapping(value = "createUser", consumes = APPLICATION_JSON_VALUE)
     public void createUser(@RequestBody UserDto userDto) {
-        users.add(userDto);
-    }
-
-    @PutMapping(value = "badUser")
-    public UserDto badUser(@RequestBody Long id) {
-        UserDto user = getUserForSpecificID(id);
-        user.setBlocked(false);
-        return user;
+        userService.saveUser(userMapper.userDtoToUser(userDto));
     }
 
-    @GetMapping(value = "getToken", consumes = APPLICATION_JSON_VALUE)
-    public Long getToken(@RequestBody Long userId) {
-        Random random = new Random();
-        Long randomLong = random.nextLong();
-        return randomLong;
+    @PutMapping(value = "blockUser")
+    public UserDto blockUser(@RequestParam Long id) throws UserNotFoundException {
+        return userMapper.userToUserDto(userService.blockUser(id));
     }
 
-    private UserDto getUserForSpecificID(Long id) {
-        for (UserDto user : users) {
-            if (user.getId() == id) {
-                return user;
-            }
-        }
-        return null;
+    @GetMapping(value = "getToken")
```

Comment:
do czego ma służyć ta metoda? jest bardzo niebezpieczna. podając ID dostajesz token jakiegokolwiek użytkownika.

Reply:
Tak było we wtepnym controllerze - ktos to klepnął ;) poprawie bo rzeczywiscie to nie ma sensu